core/core-impl/src/main/java/ai/timefold/solver/core/impl/constructionheuristic/DefaultConstructionHeuristicPhaseFactory.java

```diff
@@ -125,11 +129,26 @@ private Optional<ListVariableDescriptor<?>> findValidListVariableDescriptor(
                     + listVariableDescriptors + ").");
         }
 
+        ListVariableDescriptor<Solution_> listVariableDescriptor = listVariableDescriptors.get(0);
+        failIfBasicAndListVariablesAreCombinedOnSingleEntity(listVariableDescriptor);
         failIfConfigured(phaseConfig.getConstructionHeuristicType(), "constructionHeuristicType");
         failIfConfigured(phaseConfig.getEntityPlacerConfig(), "entityPlacerConfig");
         failIfConfigured(phaseConfig.getMoveSelectorConfigList(), "moveSelectorConfigList");
 
-        return Optional.of(listVariableDescriptors.get(0));
+        return Optional.of(listVariableDescriptor);
+    }
+
+    private static void failIfBasicAndListVariablesAreCombinedOnSingleEntity(ListVariableDescriptor<?> listVariableDescriptor) {
+        EntityDescriptor<?> listVariableEntityDescriptor = listVariableDescriptor.getEntityDescriptor();
+        if (listVariableEntityDescriptor.getDeclaredGenuineVariableDescriptors().size() > 1) {
+            Collection<GenuineVariableDescriptor<?>> basicVariableDescriptors =
+                    new ArrayList<>(listVariableEntityDescriptor.getDeclaredGenuineVariableDescriptors());
+            basicVariableDescriptors.remove(listVariableDescriptor);
+            throw new IllegalArgumentException(
+                    "Construction Heuristic phase does not support combination of basic variables ("
+                            + basicVariableDescriptors + ") and list variables (" + listVariableDescriptor
+                            + ") on a single planning entity (" + listVariableDescriptor.getEntityDescriptor() + ").");
```

Comment:
```suggestion
            throw new UnsupportedOperationException("""
                    Combining basic variables (%s) with list variables (%) on a single planning entity (%s) is currently not supported.
                    """.formatted(basicVariableDescriptors, listVariableDescriptor,  listVariableDescriptor.getEntityDescriptor()));
```

Arguably, this is not a limitation of CH, this is a limitation of the solver. Basic and list variables simply can not be combined.

Reply:
Also, arguably we don't want to put the entity descriptor into the error message, rather the full canonical name of the entity.